packages/freezed/test/equal_test.dart

```diff
@@ -139,4 +139,8 @@ void main() {
       GenericIterable([1]),
     );
   });
+  test('ObjectWithOtherProperty', () {
+    expect(ObjectWithOtherProperty(1), ObjectWithOtherProperty(1));
+    expect(ObjectWithOtherProperty(1), isNot(ObjectWithOtherProperty(2)));
```

Comment:
Could you add a test for the "DeepCollectionEquality" part too?

Reply:
Do you mean adding another class in which `other` is a `List` instead of an `int` and then adding a test for it?

Reply:
Something where the properties are not `identical`

The current test tests only the first part of the changes: `identical(other.${p.name}, this.${p.name})`

Reply:
Got it, I'll do it right away!

Reply:
What do you think about making `ObjectWithOtherProperty` take dynamic? I think that will make it easier as the class name will get too long without it.

Reply:
What's the relationship with `dynamic` and the class name?

Reply:
Nothing, I just think that using separate classes, one with `int other` and another with `List<int> other`, will make the class names to verbose. Something like this: `ObjectWithListOtherPropert`.

Reply:
There's no need for two classes. A single class that takes a `List` is enough

Reply:
But how will we test for `identical(other.${p.name}, this.${p.name}`? By making the lists `const` sometimes and sometimes not, perhaps?

Reply:
Yes

Reply:
OK, commit coming in a minutes!